lib/engine/game/base.rb

```diff
@@ -1160,10 +1160,9 @@ def check_overlap(routes)
         end
       end
 
-      def check_connected(route, token)
-        paths_ = route.paths.uniq
-
-        return if token.select(paths_, corporation: route.corporation).size == paths_.size
+      def check_connected(route, token_city)
+        corporation = token_city ? route.corporation : nil
```

Comment:
corporation = route.corporation if token_city

Reply:
and you can do token_city = nil so that's it's not required

Reply:
this method is weird now, it used to take in the token_city, but not anymore, you don't actually use the token city at all, can you clean this up?

Reply:
Yeah, I realized over night I should have done that.

Reply:
I will pass corporation instead of token to check_connected